src/Tickets/Commerce/Order.php

```diff
@@ -358,22 +359,39 @@ public static function get_flag_action_marker_meta_key( $flag, Status\Status_Int
 	 * @since 5.1.9
 	 * @since 5.18.1 Wrap status update in a transaction to prevent race conditions.
 	 *
-	 * @param int    $order_id        Which order ID will be updated.
-	 * @param string $new_status_slug Which Order Status we are modifying to.
-	 * @param array  $extra_args      Extra repository arguments.
+	 * @param int                     $order_id        Which order ID will be updated.
+	 * @param string|Status_Interface $new_status      Which Order Status we are modifying to.
+	 * @param array                   $extra_args      Extra repository arguments.
 	 *
-	 * @return bool|\WP_Error
+	 * @return bool
 	 */
-	public function modify_status( $order_id, $new_status_slug, array $extra_args = [] ) {
-		$new_status = tribe( Status\Status_Handler::class )->get_by_slug( $new_status_slug );
+	public function modify_status( $order_id, $new_status, array $extra_args = [] ): bool {
+		if ( ! $new_status instanceof Status\Status_Interface ) {
+			$new_status = tribe( Status\Status_Handler::class )->get_by_slug( (string) $new_status );
+		}
 
-		if ( ! $new_status ) {
+		if ( ! $new_status instanceof Status\Status_Interface ) {
 			return false;
 		}
 
+		$required_previous_status = $new_status->required_previous_status();
+		if ( $required_previous_status ) {
+			$order = tec_tc_get_order( $order_id );
+			foreach ( $required_previous_status as $required_status ) {
+				// If the required status is present we skip that status.
+				if ( isset( $order->status_log[ $required_status->get_slug() ] ) ) {
+					continue;
+				}
+
+				// Recursively call the modify_status method to ensure all required statuses are met.
+				$this->modify_status( $order_id, $required_status, $extra_args );
+			}
+		}
+
+		// we specifically only lock the order transactions after the required statuses have been met.
```

Comment:
The comment is not correct the order is being locked for each update to the required statuses...

Also we could end up in a loop calling this method recursively